### ISSUE TITLE
Delete store paths for environments more selectively

### ISSUE THREAD

**Issue by ysndr** [flox/flox#130]
I think `nix store gc` might actually clean up too much for the developer persona. I.e. dev shells and dependendencies will need to be recreated after a gc run / `flox wipe-history`.

The current approach works for us and most flox users but we might look closer at using `nix store delete` to delete paths more selectively.

_Originally posted by @ysndr in https://github.com/flox/flox/pull/119#discussion_r1246728170_

**Comment by mkenigs** [comment]
Agreed. https://github.com/NixOS/nix/pull/8417 could also be relevant to help with more targeted GC

**Comment by limeytexan** [comment]
Putting into icebox as we await development of this feature in Nix upstream.

**Comment by zmitchell** [comment]
We have `flox gc` now which runs a full `nix store gc`.
